test/typescript/runtime.test.ts

```diff
@@ -0,0 +1,266 @@
+import cds, { User, Query } from '@sap/cds';
+import { struct } from '@sap/cds/apis/csn';
+
+import {
+  Service,
+  EventContext,
+  Request,
+  Event,
+  // User, >> already imported above
+
+  ApplicationService,
+  MessagingService,
+  DatabaseService,
+  RemoteService,
+
+  Association,
+  Composition,
+  // entity,
+  event,
+  // type,
+  array,
+  // struct,
+  service,
+
+} from '@sap/cds';
+
+
+describe('runtime tests', () => {
+
+  let {Test} = cds.test
+  let test = new Test
+  let log = test.log()
+
+  it('cds.test', () => {
+    expect(test.test).toBe(test)
+    expect(typeof test.run).toBe('function')
+    expect(typeof test.in).toBe('function')
+    expect(typeof test.log).toBe('function')
+    expect(typeof test.data.reset).toBe('function')
+    console.log('foo')
+    expect(log.output).toBe('foo\n') // only works in jest/mocha (which trigger beforeAll)
+    console.log('bar')
+    expect(log.output).toBe('foo\nbar\n')
+    log.release()
+    console.log('car')
+    expect(log.output).toBe('')
```

Comment:
With a test runner executing the `beforeAll` hooks, the logging capture now works as expected.